From ffb78e62bc1ebc0856ac08224d47846b3d9d52b5 Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Wed, 31 Jan 2024 15:42:14 +0000
Subject: [PATCH 007/115] Mention support for AMD/znver5 in GAS

---
 gas/NEWS | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gas/NEWS b/gas/NEWS
index f19fd471c22..67d806cbf89 100644
--- a/gas/NEWS
+++ b/gas/NEWS
@@ -2,6 +2,8 @@
 
 Changes in 2.42:
 
+* Added support for AMD znver5 processor (available from GNU Binutils 2.42 release).
+
 * Add support for the AArch64 Scalable Vector Extension version 2.1 (SVE2.1).
 
 * Add support for the AArch64 Scalable Matrix Extension version 2.1 (SME2.1).
-- 
2.44.0

