From 995e0b770e46b5a319fc2591b04421531ec21566 Mon Sep 17 00:00:00 2001
From: Tamar Christina <tamar.christina@arm.com>
Date: Tue, 25 May 2021 18:41:31 +0100
Subject: [PATCH 130/135] Gas: Forgot to commit changelog

---
 gas/ChangeLog | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/gas/ChangeLog b/gas/ChangeLog
index e747d7dd8a9..0503d1beb0b 100644
--- a/gas/ChangeLog
+++ b/gas/ChangeLog
@@ -1,3 +1,14 @@
+2021-05-25  Tamar Christina  <tamar.christina@arm.com>
+
+	Backported from mainline.
+	2021-05-25  Tamar Christina  <tamar.christina@arm.com>
+
+	PR gas/25235
+	* config/tc-arm.c (md_convert_frag): Set LSB when Thumb symbol.
+	(relax_adr): Thumb symbols 4 bytes.
+	* testsuite/gas/arm/pr25235.d: New test.
+	* testsuite/gas/arm/pr25235.s: New test.
+
 2021-05-19  Alex Coplan  <alex.coplan@arm.com>
 
 	* config/tc-arm.c (do_mve_mov): Only reject vmov if we're moving
-- 
2.31.1

