--- binutils-2.29/binutils/arparse.c.omv~	2017-07-26 16:44:20.881534851 +0200
+++ binutils-2.29/binutils/arparse.c	2017-07-26 16:44:35.838762480 +0200
@@ -292,7 +292,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/binutils/sysinfo.c.omv~	2017-07-26 16:42:46.654039008 +0200
+++ binutils-2.29/binutils/sysinfo.c	2017-07-26 16:43:02.907305612 +0200
@@ -250,7 +250,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/gas/itbl-parse.c.omv~	2017-07-26 16:45:11.571297306 +0200
+++ binutils-2.29/gas/itbl-parse.c	2017-07-26 16:45:19.802418845 +0200
@@ -502,7 +502,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/gas/rl78-parse.c.omv~	2017-07-26 16:45:32.015598119 +0200
+++ binutils-2.29/gas/rl78-parse.c	2017-07-26 16:45:41.021729533 +0200
@@ -583,7 +583,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/gas/rx-parse.c.omv~	2017-07-26 16:44:55.325055640 +0200
+++ binutils-2.29/gas/rx-parse.c	2017-07-26 16:45:04.235188478 +0200
@@ -602,7 +602,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/ld/deffilep.c.omv~	2017-07-26 16:45:49.884858236 +0200
+++ binutils-2.29/ld/deffilep.c	2017-07-26 16:45:57.855973474 +0200
@@ -390,7 +390,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
--- binutils-2.29/ld/ldgram.c.omv~	2017-07-26 16:46:40.851587328 +0200
+++ binutils-2.29/ld/ldgram.c	2017-07-26 16:46:49.886714790 +0200
@@ -544,7 +544,7 @@ typedef short int yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__ && !defined __clang__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
