diff -up binutils-2.31/ld/emultempl/elf32.em.0128~ binutils-2.31/ld/emultempl/elf32.em
--- binutils-2.31/ld/emultempl/elf32.em.0128~	2018-07-18 22:15:00.159257178 +0200
+++ binutils-2.31/ld/emultempl/elf32.em	2018-07-18 22:15:00.205255988 +0200
@@ -191,7 +191,7 @@ static char *audit; /* colon (typically)
 static char *depaudit; /* colon (typically) separated list of libs */
 
 /* Style of .note.gnu.build-id section.  */
-static const char *emit_note_gnu_build_id;
+static const char *emit_note_gnu_build_id = NULL;
 
 /* On Linux, it's possible to have different versions of the same
    shared library linked against different versions of libc.  The
@@ -806,6 +806,9 @@ gld${EMULATION_NAME}_check_ld_elf_hints
       FILE *f;
       char *tmppath;
 
+      if (!emit_note_gnu_build_id)
+        emit_note_gnu_build_id = xstrdup("sha1");
+
       tmppath = concat (ld_sysroot, _PATH_ELF_HINTS, (const char *) NULL);
       f = fopen (tmppath, FOPEN_RB);
       free (tmppath);
diff -up binutils-2.31/ld/ldmain.c.0128~ binutils-2.31/ld/ldmain.c
--- binutils-2.31/ld/ldmain.c.0128~	2018-07-18 22:15:00.205255988 +0200
+++ binutils-2.31/ld/ldmain.c	2018-07-18 22:16:06.428555216 +0200
@@ -281,7 +281,10 @@ main (int argc, char **argv)
 
   link_info.allow_undefined_version = TRUE;
   link_info.keep_memory = TRUE;
+  link_info.optimize = TRUE;
+  link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
+  link_info.relro = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.prohibit_multiple_definition_absolute = FALSE;
   link_info.emit_hash = DEFAULT_EMIT_SYSV_HASH;
