--- binutils-linaro-2.25.0-2015.01-2/config/gettext.m4.omv~	2015-06-21 22:25:34.071368100 +0200
+++ binutils-linaro-2.25.0-2015.01-2/config/gettext.m4	2015-06-21 22:27:43.938986435 +0200
@@ -135,10 +135,10 @@ AC_DEFUN([AM_GNU_GETTEXT],
 changequote(,)dnl
 typedef int array [2 * (__GNU_GETTEXT_SUPPORTED_REVISION(0) >= 1) - 1];
 changequote([,])dnl
-], [])[extern int _nl_msg_cat_cntr;
-extern int *_nl_domain_bindings;],
+], [])[
+],
             [bindtextdomain ("", "");
-return (int) gettext ("")]ifelse([$2], [need-ngettext], [ + (int) ngettext ("", "", 0)], [])[ + _nl_msg_cat_cntr + *_nl_domain_bindings],
+return (int) gettext ("")]ifelse([$2], [need-ngettext], [ + (int) ngettext ("", "", 0)], []),
             gt_cv_func_gnugettext_libc=yes,
             gt_cv_func_gnugettext_libc=no)])
 
--- binutils-linaro-2.25.0-2015.01-2/intl/configure.omv~	2015-06-21 22:31:13.423777241 +0200
+++ binutils-linaro-2.25.0-2015.01-2/intl/configure	2015-06-21 22:31:24.232663770 +0200
@@ -5778,13 +5778,11 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <libintl.h>
-extern int _nl_msg_cat_cntr;
-extern int *_nl_domain_bindings;
 int
 main ()
 {
 bindtextdomain ("", "");
-return (int) gettext ("") + _nl_msg_cat_cntr + *_nl_domain_bindings
+return (int) gettext ("")
   ;
   return 0;
 }
