diff -up binutils-2.34/ld/emultempl/elf.em.11~ binutils-2.34/ld/emultempl/elf.em
diff -up binutils-2.34/ld/ldelf.c.11~ binutils-2.34/ld/ldelf.c
--- binutils-2.34/ld/ldelf.c.11~	2020-02-04 02:22:18.582746943 +0100
+++ binutils-2.34/ld/ldelf.c	2020-02-04 02:25:42.767755946 +0100
@@ -47,7 +47,7 @@ struct dt_needed
 };
 
 /* Style of .note.gnu.build-id section.  */
-const char *ldelf_emit_note_gnu_build_id;
+const char *ldelf_emit_note_gnu_build_id = NULL;
 
 /* These variables are required to pass information back and forth
    between after_open and check_needed and stat_needed and vercheck.  */
@@ -695,6 +695,9 @@ ldelf_check_ld_elf_hints (const struct b
       FILE *f;
       char *tmppath;
 
+      if (!ldelf_emit_note_gnu_build_id)
+        ldelf_emit_note_gnu_build_id = xstrdup("sha1");
+
       tmppath = concat (ld_sysroot, _PATH_ELF_HINTS, (const char *) NULL);
       f = fopen (tmppath, FOPEN_RB);
       free (tmppath);
diff -up binutils-2.34/ld/ldmain.c.11~ binutils-2.34/ld/ldmain.c
--- binutils-2.34/ld/ldmain.c.11~	2020-01-18 14:55:48.000000000 +0100
+++ binutils-2.34/ld/ldmain.c	2020-02-04 02:19:41.014739995 +0100
@@ -283,7 +283,10 @@ main (int argc, char **argv)
 
   link_info.allow_undefined_version = TRUE;
   link_info.keep_memory = TRUE;
+  link_info.optimize = TRUE;
+  link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
+  link_info.relro = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.prohibit_multiple_definition_absolute = FALSE;
   link_info.emit_hash = DEFAULT_EMIT_SYSV_HASH;
