diff -up binutils-2.31/bfd/Makefile.am.0003~ binutils-2.31/bfd/Makefile.am
--- binutils-2.31/bfd/Makefile.am.0003~	2018-07-16 10:06:05.963201333 +0200
+++ binutils-2.31/bfd/Makefile.am	2018-07-16 10:09:21.353192372 +0200
@@ -33,7 +33,7 @@ bfdlibdir = @bfdlibdir@
 bfdincludedir = @bfdincludedir@
 bfdlib_LTLIBRARIES = libbfd.la
 bfdinclude_HEADERS = $(BFD_H) $(INCDIR)/ansidecl.h $(INCDIR)/symcat.h \
-		     $(INCDIR)/bfdlink.h $(INCDIR)/diagnostics.h
+		     $(INCDIR)/bfdlink.h $(INCDIR)/diagnostics.h $(INCDIR)/demangle.h
 else !INSTALL_LIBBFD
 # Empty these so that the respective installation directories will not be created.
 bfdlibdir =
diff -up binutils-2.31/bfd/Makefile.in.0003~ binutils-2.31/bfd/Makefile.in
--- binutils-2.31/bfd/Makefile.in.0003~	2018-07-16 10:06:05.964201333 +0200
+++ binutils-2.31/bfd/Makefile.in	2018-07-16 10:09:37.094191650 +0200
@@ -248,7 +248,7 @@ am__can_run_installinfo = \
   esac
 am__bfdinclude_HEADERS_DIST = $(INCDIR)/plugin-api.h bfd.h \
 	$(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h \
-	$(INCDIR)/diagnostics.h
+	$(INCDIR)/diagnostics.h $(INCDIR)/demangle.h
 HEADERS = $(bfdinclude_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
