--- binutils-2.23.51.0.2/bfd/elf64-ppc.c.ppc64-pie~	2012-09-10 18:28:03.000000000 +0200
+++ binutils-2.23.51.0.2/bfd/elf64-ppc.c	2012-09-12 13:00:42.343395824 +0200
@@ -13523,6 +13523,9 @@ ppc64_elf_relocate_section (bfd *output_
 		{
 		  BFD_ASSERT (h->elf.dynindx != -1);
 		  outrel.r_info = ELF64_R_INFO (h->elf.dynindx, r_type);
+                  if (h->elf.dynidx == -1
+                      && h->elf.root.type == bfd_link_hash_undefweak)
+                    memset (&outrel, 0, sizeof outrel);
 		}
 	      else
 		{
