--- binutils-2.25.51/bfd/merge.c.sec-merge-emit~	2015-07-11 14:04:48.899796115 +0200
+++ binutils-2.25.51/bfd/merge.c	2015-07-11 14:06:19.496685721 +0200
@@ -308,6 +308,9 @@ sec_merge_emit (bfd *abfd, struct sec_me
       len = -off & (entry->alignment - 1);
       if (len != 0)
 	{
+          /* We should never have an entry with an alignment
+             greater than the section's alignment.  */
+          BFD_ASSERT (len <= (bfd_size_type) (1 << alignment_power));
 	  if (contents)
 	    {
 	      memcpy (contents + offset, pad, len);
@@ -334,7 +337,7 @@ sec_merge_emit (bfd *abfd, struct sec_me
 
   /* Trailing alignment needed?  */
   off = sec->size - off;
-  if (off != 0)
+  if (off != 0 && alignment_power)
     {
       if (contents)
 	memcpy (contents + offset, pad, off);
