--- binutils-2.27.90/binutils/syslex.l.yywrap~	2017-01-17 23:32:32.000000000 +0100
+++ binutils-2.27.90/binutils/syslex.l	2017-01-17 23:56:02.112615921 +0100
@@ -1,4 +1,4 @@
-%option noinput nounput
+%option noinput nounput noyywrap
 
 %{
 /* Copyright (C) 2001-2017 Free Software Foundation, Inc.
@@ -36,10 +36,6 @@
 #define YY_NO_UNPUT
 #endif
 
-#ifndef yywrap
-static int yywrap (void) { return 1; }
-#endif
-
 extern int yylex (void);
 %}
 %%
--- binutils-2.27.90/ld/ldlex.l.yywrap~	2017-01-17 23:32:34.000000000 +0100
+++ binutils-2.27.90/ld/ldlex.l	2017-01-17 23:54:57.809163173 +0100
@@ -1,4 +1,4 @@
-%option nounput
+%option nounput noyywrap
 
 %{
 
@@ -86,10 +86,6 @@ static void lex_warn_invalid (char *wher
 */
 #define RTOKEN(x)  {  yylval.token = x; return x; }
 
-/* Some versions of flex want this.  */
-#ifndef yywrap
-int yywrap (void) { return 1; }
-#endif
 %}
 
 %a 4000
