diff -up binutils-2.34.90/ld/ldelf.c.10~ binutils-2.34.90/ld/ldelf.c
--- binutils-2.34.90/ld/ldelf.c.10~	2020-07-16 04:28:48.000000000 +0200
+++ binutils-2.34.90/ld/ldelf.c	2020-07-16 04:44:49.616141952 +0200
@@ -47,7 +47,7 @@ struct dt_needed
 };
 
 /* Style of .note.gnu.build-id section.  */
-const char *ldelf_emit_note_gnu_build_id;
+const char *ldelf_emit_note_gnu_build_id = NULL;
 
 /* These variables are required to pass information back and forth
    between after_open and check_needed and stat_needed and vercheck.  */
@@ -700,6 +700,9 @@ ldelf_check_ld_elf_hints (const struct b
       FILE *f;
       char *tmppath;
 
+      if (!ldelf_emit_note_gnu_build_id)
+        ldelf_emit_note_gnu_build_id = xstrdup("sha1");
+
       tmppath = concat (ld_sysroot, _PATH_ELF_HINTS, (const char *) NULL);
       f = fopen (tmppath, FOPEN_RB);
       free (tmppath);
diff -up binutils-2.34.90/ld/ldmain.c.10~ binutils-2.34.90/ld/ldmain.c
--- binutils-2.34.90/ld/ldmain.c.10~	2020-07-16 04:44:49.616141952 +0200
+++ binutils-2.34.90/ld/ldmain.c	2020-07-16 04:45:42.418701923 +0200
@@ -331,7 +331,10 @@ main (int argc, char **argv)
 
   link_info.allow_undefined_version = TRUE;
   link_info.keep_memory = TRUE;
+  link_info.optimize = TRUE;
+  link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
+  link_info.relro = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.prohibit_multiple_definition_absolute = FALSE;
   link_info.textrel_check = DEFAULT_LD_TEXTREL_CHECK;
